### PR TITLE
xds: minor refactor introducing an interface for GrpcStream

### PR DESCRIPTION
**Pull request by adisuissa**
Commit Message: xds: minor refactor introducing an interface for GrpcStream
Additional Description:
Introducing an interface for GrpcStream to make it easier to test (as part of the work of xDS-Failover support #28099)

Risk Level: low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: source/extensions/config_subscription/grpc/grpc_stream_interface.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "source/common/grpc/typed_async_client.h"
+
+namespace Envoy {
+namespace Config {
+
+// Oversees communication for gRPC xDS implementations (parent to both regular xDS and delta
```
Comment (abeyad) [inline source/extensions/config_subscription/grpc/grpc_stream_interface.h:8]: nit: s/regular/state-of-the-world ?

File: source/extensions/config_subscription/grpc/grpc_stream_interface.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// xDS variants). Reestablishes the gRPC channel when necessary, and provides rate limiting of
+// requests. This interface is for a specific discovery Request and Response types.
```
Comment (abeyad) [inline source/extensions/config_subscription/grpc/grpc_stream_interface.h:10]: what does "specific discovery Request and Response types" mean here exactly?

**Review by abeyad — DISMISSED** [review]
Looks great, thanks!  Left a minor optional nit and a question.

File: source/extensions/config_subscription/grpc/grpc_stream_interface.h (hunk 1 above)
Comment (adisuissa) [reply source/extensions/config_subscription/grpc/grpc_stream_interface.h]: Done.

File: source/extensions/config_subscription/grpc/grpc_stream_interface.h (hunk 2 above)
Comment (adisuissa) [reply source/extensions/config_subscription/grpc/grpc_stream_interface.h]: Agree that this is confusing, removed.

**Review by abeyad — APPROVED** [review]

**Review by htuch — APPROVED** [review]

**Comment by adisuissa** [comment]
Alex will you be able to do a cross-company review?
/assign @soulxu

**Review by soulxu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**